File: src/config/persistence.py
from __future__ import annotations

import tomllib
import toml

from config.settings import DocumentModel


def save_model(model: DocumentModel, path: str) -> None:
    data = model.to_dict()
    with open(path, "w", encoding="utf-8") as f:
        toml.dump(data, f)


def load_model(path: str) -> DocumentModel:
    with open(path, "rb") as f:
        data = tomllib.load(f)
    return DocumentModel.from_dict(data)

File: src/main/persistence.ts
/**
 * TOML 配置持久化（移植自 src/config/persistence.py）。
 * 用 smol-toml 替代 Python `toml`/`tomllib`。
 */
import * as fs from 'node:fs'
import { stringify, parse } from 'smol-toml'
import {
  documentToDict,
  documentFromDict,
  type DocumentModel
} from '@shared/settings'

export function saveModel(model: DocumentModel, filePath: string): void {
  const data = documentToDict(model)
  const toml = stringify(data as Record<string, unknown>)
  fs.writeFileSync(filePath, toml, 'utf-8')
}

export function loadModel(filePath: string): DocumentModel {
  const raw = fs.readFileSync(filePath, 'utf-8')
  const data = parse(raw) as Record<string, unknown>
  return documentFromDict(data)
}
